bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/handler/VerisureAlarmThingHandler.java

```diff
@@ -0,0 +1,164 @@
+import org.openhab.binding.verisure.internal.model.VerisureThing;
+
+/**
+ * Handler for the Alarm Device thing type that Verisure provides.
+ *
+ * @author Jan Gustafsson - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class VerisureAlarmThingHandler extends VerisureThingHandler {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_ALARM);
+
+    private static final int REFRESH_DELAY_SECONDS = 10;
+
+    public VerisureAlarmThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("handleCommand, channel: {}, command: {}", channelUID, command);
+        if (command instanceof RefreshType) {
+            super.handleCommand(channelUID, command);
+        } else if (channelUID.getId().equals(CHANNEL_ALARM_STATUS)) {
+            handleAlarmState(command);
+            scheduleImmediateRefresh(REFRESH_DELAY_SECONDS);
+        } else {
+            logger.warn("Unknown command! {}", command);
+        }
+    }
+
+    private void handleAlarmState(Command command) {
+        String deviceId = config.getDeviceId();
+        if (session != null && deviceId != null) {
+            VerisureAlarms alarm = (VerisureAlarms) session.getVerisureThing(deviceId);
+            if (alarm != null) {
+                BigDecimal installationId = alarm.getSiteId();
+                String pinCode = session.getPinCode(installationId);
+
+                if (pinCode != null && installationId != null) {
+                    StringBuilder sb = new StringBuilder(deviceId);
+                    sb.insert(4, " ");
+                    String url = START_GRAPHQL;
+                    String operation, state = "";
+
+                    switch (command.toString()) {
+                        case "DISARMED":
+                            operation = "disarm";
+                            state = "armStateDisarm";
+                            break;
+                        case "ARMED_HOME":
+                            operation = "armHome";
+                            state = "armStateArmHome";
+                            break;
+                        case "ARMED_AWAY":
+                            operation = "armAway";
+                            state = "armStateArmAway";
+                            break;
+                        default:
+                            logger.warn("Unknown alarm command: {}", command);
+                            return;
+                    }
+
+                    String queryQLAlarmSetState = "[{\"operationName\":\"" + operation + "\",\"variables\":{\"giid\":\""
+                            + installationId + "\",\"code\":\"" + pinCode + "\"},\"query\":\"mutation " + operation
+                            + "($giid: String!, $code: String!) {\\n  " + state
+                            + "(giid: $giid, code: $code)\\n}\\n\"}]\n" + "";
```

Comment:
Please use a java object instead and use gson to convert it to json.

Reply:
Brilliant, why didn't I think of that, I'll fix!

Reply:
I've actually implemented this for VerisureSession too that use JSON for polling status aswell, turned out much better. Great finding! :-)